File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let result = 0;
    let lowest = prices[0];

    for (price of prices) {
    lowest = Math.min(lowest,price)
    result = Math.max(result, price - lowest)
    }
    return result;
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.rs
impl Solution {
    pub fn max_profit(prices: Vec<i32>) -> i32 {
        let mut result = 0;
        let mut lowest = prices[0];
        for price in prices {
            lowest = std::cmp::min(lowest,price);
            result = std::cmp::max(result, price - lowest);
        }
        return result;
    }
}

